pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskExecutor.java

```diff
@@ -46,14 +46,14 @@ protected SegmentConversionResult convert(PinotTaskConfig pinotTaskConfig, File
     String rawTableName = TableNameBuilder.extractRawTableName(tableNameWithType);
 
     SegmentPurger.RecordPurgerFactory recordPurgerFactory = MINION_CONTEXT.getRecordPurgerFactory();
+    TableConfig tableConfig = getTableConfig(tableNameWithType);
+    Schema schema = getSchema(tableNameWithType);
     SegmentPurger.RecordPurger recordPurger =
-        recordPurgerFactory != null ? recordPurgerFactory.getRecordPurger(rawTableName) : null;
+        getRecordPurger(pinotTaskConfig, recordPurgerFactory, tableConfig, schema);
```

Comment:
Nit: Maybe the ternary operator reads better, and it's in sync with recordModifer initialization?

Reply:
+1 to Sajjad's comment about ternary operator reads better.

Reply:
Removed the method and kept the ternary operator